R/otp-setup.R

```diff
@@ -0,0 +1,290 @@
+#' @param memory A positive integer. Amount of memory to assign to the OTP in GB, default is 2
+#' @param router A character vector for the name of the routers, must match with contents of dir, default "current"
+#' Only a single router is currently supported
+#' @param port A positive integer. Optional, default is 8080.
+#' @param secure_port A positive integer. Optional, default is 8081.
+#' @param analyist Logical. Should the analyist features be loaded? Default FALSE
+#' @param wait Logical, Should R wait until OTP has loaded before running next line of code, default TRUE
+#' @return
+#' This function does not return a value to R.
+#' If wait is TRUE R will wait until OTP is running (maximum of 5 minutes)
+#' @examples
+#' otp_setup("C:/otp","C:/data")
+#' otp_setup("C:/otp","C:/data", memory = 5, analyst = TRUE)
+#' @export
+otp_setup <- function(otp = NULL,
+                      dir = NULL,
+                      memory = 2,
+                      router = "current",
+                      port = 8080,
+                      securePort = 8081,
+                      analyst = FALSE,
+                      wait = TRUE)
+{
+  # Run Checks
+  jar_file <- otp_checks(otp = otp, dir = dir, router = router, graph = T)
+  memory <- floor(memory) # Can have fractions of GB
+
+
+  # Set up OTP
+  if(Sys.info()[['sysname']] == "Linux") {
+    message("You're on linux, this function is not yet supported")
+    stop()
+
+  }else if(Sys.info()[['sysname']] == "Windows"){
+
+    text <- paste0('java -Xmx',memory,'G -jar "',
+                   otp,'/',jar_file,
+                   '" --router ',router,
+                   ' --graphs "',dir,'/graphs"',
+                   ' --server --port ',port,
+                   ' --securePort ',securePort)
+
+    if(analyst){
+      text <- paste0(text," --analyst")
+    }
+    set_up <- try(system(text, intern = FALSE, wait = FALSE))
+
+
+  }else if(Sys.info()[['sysname']] == "Darwin"){
+    message("You're on Mac, this function is not yet supported")
+    stop()
+
+  }else{
+    message("You're on and unknow OS, this function is not yet supported")
+    stop()
+
+  }
+
+
+  # Check for errors
+  if(grepl("ERROR",set_up[2])){
+    message("Failed to build graph with message:")
+    message(set_up[2])
+  }
+
+  message(paste0(Sys.time()," OTP is loading and may take a while to be useable"))
+
+  if(wait){
+    Sys.sleep(30)
+
+    # Check if connected
+    for(i in 1:10){
+      #message(paste0("Attempt ",i))
+      otpcon <- try(otp_connect(hostname = "localhost",
+                                router = router,
+                                port = port,
+                                ssl = FALSE,
+                                check = TRUE), silent = T)
+      if("otpconnect" %in% class(otpcon)){
+        message(paste0(Sys.time()," OTP is ready to use Go to localhost:",port," in your browser to view the OTP"))
+        browseURL(paste0(ifelse(otpcon$ssl,"https://","http://"),"localhost:",port))
```

Comment:
@mem48

Best to use the make_url method from otp_connect() here.

Reply:
good call